app/controllers/EHealthVDCController.scala

```diff
@@ -1,144 +1,202 @@
+
+    val data = Json.obj(
+      "query" -> query,
+      "purpose" -> purpose,
+      "access" -> "read",
+      "requester" -> "",
+      "blueprintId" -> "",
+      "requesterId" -> requesterId
+    )
+    val futureResponse = ws.url(url).addHttpHeaders("Content-Type" -> "application/json")
+      .addHttpHeaders("Accept" -> "application/json").withRequestTimeout(Duration.Inf).post(data)
+
+    val res = Await.result(futureResponse, Duration.Inf)
+    res.body[String]
   }
 
   @ApiOperation(nickname = "getAllValuesForBloodTestComponent",
     value = "Get timeseries of patient's blood test component",
     notes =  "This method returns the collected values for a specific blood test component of a patient (identified by his SSN), to be used by medical doctors",
-    response = classOf[models.Patient], responseContainer = "List", httpMethod = "GET")
+    response = classOf[models.BloodTestComponentValue], responseContainer = "List", httpMethod = "GET")
   @ApiResponses(Array(
     new ApiResponse(code = 404, message = "Patient not found")))
-  def getTestValues(@ApiParam(value = "SSN", required = true,
-    allowMultiple = false) socialId: String,
-                    @ApiParam(value = "component", required = true,
-                      allowMultiple = false) testType: String): Action[AnyContent] = {
-    Action.async {
+  def getAllValuesForBloodTestComponent(@ApiParam(value = "SSN", required = true, allowMultiple = false) socialId: String,
+                                        @ApiParam(value = "component", required = true,
+                                          allowMultiple = false) testType: String)= Action.async {
+    implicit request =>
       val spark = initService.getSparkSessionInstance
-      readData(spark)
-      val query = "select patientId, date, %s.value as %s from joined where socialId='%s'".format(testType, testType, socialId)
-      val patientBloodTestsDF = spark.sql(query)
-      patientBloodTestsDF.limit(10).show(false)
-      patientBloodTestsDF.printSchema
-       patientBloodTestsDF.explain(true)
-
-      val rawJson = patientBloodTestsDF.limit(10).toJSON.collect().mkString
-      Future.successful(Ok(Json.toJson(rawJson)))
-    }
+      val patientSSN = socialId
+      var origtestType = testType
+
+      debugMode = initService.getDebugMode
+      if (config.has("policy.enforcement.play.url")) {
+        val url: String = config.get[String]("policy.enforcement.play.url")
+        val response = sendRequestToEnforcmentEngine(request.headers("Purpose"),
+          request.headers("RequesterId"), url, origtestType)
+
+        val queryOnJoinTables = "SELECT patientId, date, %s FROM joined WHERE socialId=\"%s\"".format(testType,
+          patientSSN)
+        var resultDF = getCompilantResult(spark, response, config, queryOnJoinTables)
+
+        //Adjust output to blueprint
+        resultDF = resultDF.withColumnRenamed(testType, "value").drop(Constants.SUBJECT_ID_COL_NAME).distinct()
```

Comment:
You won't need it if you use the join as in https://docs.databricks.com/spark/latest/faq/join-two-dataframes-duplicated-column.html